digital_land/model/entry.py

```diff
@@ -17,17 +18,28 @@ def __init__(self, data: Item, resource: str, line_num: int = None):
 
         self.slug = data["slug"]
 
+        if not data.get("entry-date", None):
+            raise ValueError("Entry missing entry-date")
+
+        self.entry_date = datetime.fromisoformat(data.pop("entry-date"))
+
     def __hash__(self):
         return hash((self.resource, self.line_num, frozenset(self.data.items())))
 
     @classmethod
     def from_facts(
-        cls, entity: str, facts: Set[Fact], resource: str, line_num: int = None
+        cls,
+        entity: str,
+        facts: Set[Fact],
+        resource: str,
+        line_num: int,
```

Comment:
There's no longer a default for `line_num` - was that intentional?

Reply:
yes - we shouldn't see missing line numbers unless there's a bug.